File: Laboratory/Integration/decode.ts
import {Library, ILibrary, IAlbum} from "./schema";
import {makelib} from "./makelib";
import fs from "fs";
import equal from "deep-equal";
import {assert} from "console";
import util from "util";

const buffer = fs.readFileSync(process.argv[2]);
const de = Library.decode(buffer);
const ex = makelib();

const eq = equal(de, ex, {strict: false});
if (!eq) {
    console.log("decoded:")
    console.log(util.inspect(de, {showHidden: false, depth: null}))
    console.log()
    console.log("expected:")
    console.log(util.inspect(ex, {showHidden: false, depth: null}))
}

process.exit(eq ? 0 : 1);

File: Laboratory/Integration/Python/src/decode.py
from schema import Library
from lib import make_lib
import sys
import json

with open(sys.argv[1], "rb") as buffer_file:
    buffer = buffer_file.read()

    de = Library.decode(buffer)
    ex = make_lib()

    eq = json.loads(repr(de)) == json.loads(repr(ex))

    if not eq:
        print("decoded:")
        print(repr(de))
        print()
        print("expected:")
        print(repr(ex))

    sys.exit(0 if eq else 1)